tasks/main-clevis.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Ensure required packages are installed
+  package:
+    name: "{{ __nbde_client_package_list }}"
+    state: present
+
+
+- name: Prepare keyfiles, perform clevis operations and dispose of keyfiles
+  block:
+    - name: Ensure we transfer keyfiles
+      copy:
+        src: "{{ item.keyfile | default('/dev/null') }}"
+        dest: "{{ __nbde_client_managed_dir }}/"
+        mode: '0400'
+      loop: "{{ nbde_client_bindings }}"
+
+    - name: Perform clevis operations
+      nbde_client_clevis:
+        bindings: "{{ nbde_client_bindings | default([]) }}"
+        data_dir: "{{ __nbde_client_managed_dir }}"
+      notify: nbde_client update initramfs
+
+  always:
+    - name: Ensure we dispose of any transferred keyfiles
+      file:
+        path: "{{ __nbde_client_managed_dir }}/{{ item.keyfile | default('/dev/null') | basename }}"  # yamllint disable-line rule:line-length
```

Comment:
maybe skip this task 
```yaml
  when:
    - nbde_client_bindings | d([])
    - item.keyfile | d('')
```
?

Reply:
Yeah, this should be perfect, to skip the task if there are no keyfiles to copy. Let me try this.

Reply:
Skipping here and in the copy task also, thanks.